src/connectors/autocomplete/__tests__/connectAutocomplete-test.ts

```diff
@@ -813,6 +818,58 @@ search.addWidgets([
       });
     });
 
+    it('does not send view event when hits are stalled rendered', async () => {
+      const renderFn = jest.fn();
+      const makeWidget = connectAutocomplete(renderFn);
+      const widget = makeWidget({});
+
+      const hits = [
+        {
+          objectID: '1',
+          fake: 'data',
+          __queryID: 'test-query-id',
+          __position: 0,
+        },
+        {
+          objectID: '2',
+          sample: 'infos',
+          __queryID: 'test-query-id',
+          __position: 1,
+        },
+      ];
+
+      const instantSearchInstance = instantsearch({
+        searchClient: createSearchClient({
+          search() {
+            return Promise.resolve(
+              createMultiSearchResponse(createSingleSearchResponse({ hits }))
+            );
+          },
+        }),
+        indexName: 'indexName',
+      });
+      instantSearchInstance.sendEventToInsights = jest.fn();
+      instantSearchInstance.start();
+
+      instantSearchInstance.addWidgets([widget]);
+
+      await wait(0);
+
+      expect(instantSearchInstance.sendEventToInsights).toHaveBeenCalledTimes(
+        1
+      );
+
+      instantSearchInstance.mainHelper!.hasPendingRequests = () => true;
+      instantSearchInstance._isSearchStalled = true;
+      instantSearchInstance.scheduleRender();
```

Comment:
Could we make a higher-level test that makes use of [`stalledSearchDelay`](https://www.algolia.com/doc/api-reference/widgets/instantsearch/js/#widget-param-stalledsearchdelay) and a search client mock that resolves after this time?

The test would be easier to understand, and more resilient over time.

Reply:
Yes, that could also work.

What bothers me in this test is that it interacts with deep internals and doesn't completely reflect the user scenario.

Reply:
solved with a "dead" search client in https://github.com/algolia/instantsearch.js/pull/5058/commits/7131b4a1b67ee37b0317e4531fdfec40ee121793

Thanks for the idea!